runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -213,13 +213,12 @@ public void resetTableData(CorfuRuntime runtime) {
                 .type(getUnderlyingType())
                 .build();
 
-        corfuTable.close();
-
-        Object tableObject = runtime.getObjectsView().getObjectCache().remove(oid);
+        Object tableObject = runtime.getObjectsView().getObjectCache().get(oid);
```

Comment:
I think I got your question on why we need to do throw this exception :).
I think we can even throw an exception earlier if they attempt to freeTableData() on something that was not opened and avoid doing a object cache lookup here.